### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by DanRoscigno**
Add a prompt to the PR template to encourage adding information to go into the docs.

### Changelog category (leave one):
- Documentation (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -18,5 +18,11 @@ tests/ci/run_check.py
 ### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
 ...
 
+### Documentation entry (a user-readable short description of the changes that goes to docs.clickhouse.com):
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md:21]:
```suggestion
### Documentation entry (for user-visible features and improvements, a short description of the changes that goes to docs.clickhouse.com):
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -18,5 +18,11 @@ tests/ci/run_check.py
...
+Motivation: Why is this function, tabe engine, etc. useful to ClickHouse users?
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
Motivation: Why is this SQL clause, function, table engine, etc. useful to ClickHouse users?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -18,5 +18,11 @@ tests/ci/run_check.py
...
+
+Parameters: If the feature being added takes arguments, please list the arguments and options.
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion
Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -18,5 +18,11 @@ tests/ci/run_check.py
...
+
+Example use: A query or command.
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md:26]:
```suggestion
Example use: A query or command.

Alternatively, you can directly edit documentation source files in the "docs" folder with the same pull-request as code changes.
```

**Comment by DanRoscigno** [comment]
@alexey-milovidov do you want to change this, or add something in here about adding the doc info to the system tables?

**Review by alexey-milovidov — APPROVED** [review]
Ok, let's try.

**Comment by alexey-milovidov** [comment]
@DanRoscigno previous time it did not work.

It was a link to a page-long text with philosophical observations about the meanings of documentation.
For about one year, no one (including me) read it:

https://github.com/ClickHouse/ClickHouse/commit/6a0ba9eeb247e3db5e6ad2fe101d601ca93d5745

**Comment by blinkov** [comment]
Then let's also add a CI check that will fail if there's no diff in /docs and this section is empty?)

**Comment by DanRoscigno** [comment]
> Then let's also add a CI check that will fail if there's no diff in /docs and this section is empty?) At least for the "feature" PR category for starters, but "improvements" can be easily user-visible too.

The problem with CI checks is that they verify quantity, not quality.  There is a CI check for certain PRs that ensures that documentation is added to the `description` field in the `system.functions` table when a function is modified in a PR, but what gets added is inconsistent -- sometimes it is wonderful, and sometimes it is basically the name of the function.

Tests are an example of something that works well (in my opinion). How is test quality enforced?  I regularly see people ask for more tests in PRs, which indicates to me that tests are important to all of us; how do we make docs important to all of us?

**Comment by blinkov** [comment]
It didn't meant to be a silver bullet, I just guessed that editing inconsistent documentation is easier than a missing one :)

**Pull request merged**